Terminal.Gui/Input/Mouse/MouseBindings.cs

```diff
@@ -0,0 +1,240 @@
+﻿#nullable enable
+namespace Terminal.Gui;
+
+/// <summary>
+///     Provides a collection of <see cref="MouseBinding"/> objects bound to a combination of <see cref="MouseFlags"/>.
+/// </summary>
+/// <seealso cref="View.MouseBindings"/>
+/// <seealso cref="Command"/>
+public class MouseBindings
```

Comment:
I tried and failed. It's one of the reasons I wanted your eyes on this because I know you have a firmer grasp on generics than I.

Here's the challenge:

- I want the "add" syntax to be as simple as `KeyBinding.Add (Key.X, Command.Accept)` and `MouseBindings.Add (MouseFlags.Button1Clicked, Command.Accept)`
 
1) `KeyBindings` holds `Key` and keys off `KeyCode` (effectively). Requires `Target` to be set in the constructor so the above syntax works. 
2) `MouseBindings` is keyed off the `MouseFlags` enum.  However, context needs to be passed as `MouseEventArgs`. Thus the type in the dict is `MouseEventArgs`. No `Target` is needed.  

(Out of time... will post more of what I tried later).

Reply:
I had a hack around but also am time limited but this looks like it might work:

https://github.com/tig/Terminal.Gui/pull/43/files

I pulled a couple methods up to the base for mouse and it looked promising, let me know if you have issues continuing down that road or think that what we have here is better.